kdc/kerberos5.c

```diff
@@ -2745,6 +2798,22 @@ _kdc_as_rep(astgs_request_t r)
 	}
     }
 
+    /*
+     * Add SUPPORTED_ENCTYPES if the canonicalize bit is set. Until
+     * support is added for KerbSupportedEncryptionTypes and we can
+     * indicate what encryption types a given principal supports, we
+     * do this only for an AS-REQ exchange with the TGS.
+     */
+    if (f.canonicalize && is_tgs) {
```

Comment:
What's the rationale behind testing for `f.canonicalize`? To not send this to non-Windows clients? Also, why for TGS requests only? 

[MS-KILE] 3.3.5.6 just says:

>The KDC SHOULD<50> return in the encrypted part of the AS-REP message a PA-DATA structure with padata-type set to PA-SUPPORTED-ENCTYPES [165] (section 2.2.8), to indicate what encryption types (section 2.2.7) are supported by the KDC, and whether Claims or FAST are supported.<51>

Reply:
From testing, I found that Windows only seems to add this padata type if the `canonicalize` bit is set, so this was really just to match Windows. This only being for TGS requests was to avoid having to fetch the msDS-SupportedEncryptionTypes attribute, until proper integration for it was added.